src/icp_canister_backend/src/lib.rs

```diff
@@ -128,4 +128,25 @@ pub async fn deposit(user: Principal, timelock: u64) -> Result<(), types::Deposi
     Ok(())
 }
 
+#[ic_cdk::update]
+pub fn withdraw(deposit_id: u64) -> Result<Nat, types::WithdrawError> {
+    let caller = ic_cdk::api::caller();
+    let now = ic_cdk::api::time();
+    let maybe_deposit = DEPOSITS.with(|deposits| deposits.borrow_mut().remove(&deposit_id));
```

Comment:
Make a reverse logic.
First, get a deposit struct.
Check the deposit struct fields.
Then modify deposits map.
Then withdraw deposit.